### PR TITLE
Add list entry gateway RPC to daemon

### PR DESCRIPTION
**Pull request by octol**
Add RPC to `nym-vpnd` for listing the entry gateways

### CODE REVIEW

File: nym-vpn-core/nym-vpnd/src/command_interface/connection_handler.rs [hunk 1]
```diff
@@ -110,4 +111,15 @@ impl CommandInterfaceConnectionHandler {
...
+
+    pub(crate) async fn handle_list_entry_gateways(
+        &self,
+    ) -> Result<Vec<nym_vpn_api_client::Gateway>, VpnApiClientError> {
+        // This assumes mainnet.
+        //
+        // TODO: handle other networks, which which we probably have to resort to querying
+        // nym-api for the gateways.
+        let user_agent = nym_vpn_lib::nym_bin_common::bin_info_local_vergen!().into();
+        nym_vpn_api_client::get_entry_gateways(user_agent).await
```
Comment (octol) [inline nym-vpn-core/nym-vpnd/src/command_interface/connection_handler.rs:123]: @zaneschepke

**Review by pronebird — COMMENTED** [review]
Looks sane.

File: proto/nym/vpn.proto [hunk 2]
```diff
@@ -235,6 +235,40 @@ message ImportError {
...
+  bool can_connect = 1;
+  bool can_route_ip_v4 = 2;
+  bool can_route_ip_v6 = 3;
+  bool can_route_ip_external_v4 = 4;
+  bool can_route_ip_external_v6 = 5;
+}
+
+message ProbeOutcome {
+  AsEntry as_entry = 1;
+  AsExit as_exit = 2;
```
[the remark this answers is not in the document]
Comment (octol) [reply proto/nym/vpn.proto:253]: Yep, at least for mixnet mode

[the remark this answers is not in the document]
Comment (doums) [reply proto/nym/vpn.proto:253]: Ok, so another question: Is it expected for both of these fields to be set, or only one (depending on the "parent" message `EntryGateway` and `ExitGateway`)?

If only one of this field is expected to be set, the recommended protobuf way to design this is to use `oneof`

```
message ProbeOutcome {
  oneof probe {
    AsEntry as_entry = 1;
    AsExit as_exit = 2;
  }
}
```

https://protobuf.dev/programming-guides/proto3/#oneof

[the remark this answers is not in the document]
Comment (octol) [reply proto/nym/vpn.proto:253]: Yep both are expected to be set

File: nym-vpn-core/crates/nym-vpn-api-client/src/responses.rs [hunk 3]
```diff
@@ -104,7 +104,7 @@ fn is_recently_updated(last_updated_utc: &str) -> bool {
 }
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct Country(String);
+pub struct Country(pub String);
```
[the remark this answers is not in the document]
Comment (octol) [reply nym-vpn-core/crates/nym-vpn-api-client/src/responses.rs:107]: Thanks, fixed in https://github.com/nymtech/nym-vpn-client/pull/761 :)

**Review by neacsu — APPROVED** [review]

**Pull request merged**